tests/end2end/data/fileselect/static_fileselector.sh

```diff
@@ -0,0 +1,3 @@
+for file in $1; do
+    echo $file >> $2
+done
```

Comment:
It'd be great to do this in Python instead (perhaps via `shutils.copyfileobj`) so that running the tests on Windows doesn't require a working `bash` install.

There is already a `py_proc` fixture you can use (see e.g. `tests/unit/commands/test_userscripts.py`) where you can pass Python code as a string, and you get a list like `["python3", "-c", "yourcode"]` back which you can pass as process directly.

Reply:
Sounds good! I tried using the `py_proc` fixture as follows:
```python
FILE_SELECTOR_SCRIPT = r"""
import sys
tmp_file = sys.argv[1]
choosenfiles = {}
with open(tmp_file, 'w') as f:
    for choosenfile in choosenfiles:
        f.write(choosenfile)
"""

@bdd.when(bdd.parsers.parse('I set up a fake single file fileselector '
                            'selecting "{choosefile}"'))
def set_up_single_fileselector(quteproc, py_proc, choosefile):
    """Set up fileselect.single_file.command to select the file."""
    cmd, args = py_proc(FILE_SELECTOR_SCRIPT.format(repr([choosefile])))
    fileselect_cmd = json.dumps([cmd, *args, '{}'])
    quteproc.set_setting('fileselect.handler', 'external')
    quteproc.set_setting('fileselect.single_file.command', fileselect_cmd)
```

However, I get the error `qutebrowser.api.cmdutils.CommandError: Invalid value '...' - may not contain unprintable chars!`. Is this because of newlines or single quotes maybe? Sorry if I perhaps misunderstood what you intended :)

Reply:
Hmm, weird. Should work on first sight. Perhaps I should rewrite `py_proc` so it uses a temporary `.py` file with more complicated multiline input. Can you push it so I can take a closer look?

Reply:
Pushed, see e241c8c49

Reply:
Btw, I also restructured some things in 1b2fd8b in order to reuse code. However, the functionality is the same and the issue is still there